### DISCUSSION TITLE
Why proprierty code in this repo?

### DISCUSSION THREAD

**Discussion by drsounds** [winglang/wing#6127] [Ideas]
I found this exciting but it sucks the console thing is proprierty. This makes this amazing project no go for me. I might be radical but an open source project must be 100 % open source otherwise it is weird. At least proprierty code shouldn't be mixed into the same repo as MIT-licensed

**Comment by eladb** [comment]
@drsounds thanks for the feedback, we really appreciate it, and it's timely because this is something we have been discussing internally as well. We will update on this issue as we make some decisions.

**Comment by drsounds** [comment]
I am caring for your legal integrity as open source repos are to be forked, and it could accidently lead to unintentional license violations by contributors if proprierty code is mixed with mit licensed in same repo...

**Comment by eladb** [comment]
> I am caring for your legal integrity as open source repos are to be forked

This is a strong argument.

**Comment by eladb** [comment]
@drsounds letting you know this is still in discussions

**Reply by a-priestley** [reply]
Wanted to quickly add to this discussion to mention that I have a PR open on the mason registry in hopes of integrating your language server.

Re: [discussion:](https://github.com/mason-org/mason-registry/pull/8331#issuecomment-2559875496)
I would agree that the license mixing practice is confusing. When I determined that the "MIT" license should be specified in the definition, the reason for doing so was that it is referenced on the [npmjs](https://www.npmjs.com/package/winglang) page.

I am still waiting to see how the maintainers respond, but I would guess that we will need to add "proprietary" to the list of values here.
